cluster/router/router.go

```diff
@@ -50,3 +55,39 @@ type PriorityRouter interface {
 	// 0 to ^int(0) is better
 	Priority() int64
 }
+
+// Poolable caches address pool and address metadata for a router instance which will be used later in Router's Route.
+type Poolable interface {
+	// Pool created address pool and address metadata from the invokers.
+	Pool([]protocol.Invoker) (AddrPool, AddrMetadata)
+
+	// ShouldPool returns if it should pool. One typical scenario is a router rule changes, in this case, a pooling
+	// is necessary, even if the addresses not changed at all.
+	ShouldPool() bool
```

Comment:
maybe u should use "Enable" as this func's name. If u add this interface or func as the Dubbo Interface, u can omit my suggestion.

Reply:
ShouldPool is more accurate in my opinion. If a router receives a changed rule, it should re-pool the whole address list into bitmaps. That the reason why the router chain needs to ask each "Poolable" router should pool the address.